pkg/controller/node/nodecontroller.go

```diff
@@ -324,6 +330,19 @@ func (nc *NodeController) Run(period time.Duration) {
 	go wait.Until(nc.cleanupOrphanedPods, 30*time.Second, wait.NeverStop)
 }
 
+func (nc *NodeController) filterOutServiceRange() {
+	if !nc.clusterCIDR.Contains(nc.serviceCIDR.IP) && !nc.serviceCIDR.Contains(nc.clusterCIDR.IP) {
+		return
+	}
+
+	for i := 0; i< nc.cidrAllocator.MaxCIDRs(); i++ {
```

Comment:
In this case, the CIDR allocator is a range of node CIDRs within a cluster CIDR, though, isn't it? They're somewhat well bounded right now, e.g. 32k node CIDRs in a /9 cluster CIDR. Having started to write this PR, this is a similar approach as I took because the logic for going the bitmask exactly right, while fun, takes some crunching (and would take some thorough unit tests), whereas this is plain, slow and minorly hurts startup time only (but 32k is tiny).